mongodbatlas.go

```diff
@@ -180,13 +196,23 @@ func (m *MongoDBAtlas) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRe
 		}
 	}
 
-	// Default to "admin" if no db provided
-	if databaseUser.DatabaseName == "" {
-		databaseUser.DatabaseName = "admin"
+	// If the user is an X.509 user, delete the user from the X.509 certificate subject field
+	if strings.HasPrefix(req.Username, "CN=") {
+		if databaseUser.DatabaseName == "" {
+			databaseUser.DatabaseName = "$external"
+		}
+	} else {
+		// If the user is not an X.509 user, delete the user from the MongoDB Atlas project
+		if databaseUser.DatabaseName == "" {
+			databaseUser.DatabaseName = "admin"
+		}
 	}
 
 	_, err = client.DatabaseUsers.Delete(ctx, databaseUser.DatabaseName, m.ProjectID, req.Username)
-	return dbplugin.DeleteUserResponse{}, err
+	if err != nil {
+		return dbplugin.DeleteUserResponse{}, fmt.Errorf("error deleting user from project: %w", err)
```

Comment:
I know we have always returned the error here. But I am wondering if this error return leaks any sensitive information?

Reply:
I just tested this and I think it is ok. Here is the response I got when changing to an unknown user for example:

```
error deleting user from project: DELETE https://cloud.mongodb.com/api/atlas/v1.0/groups/5f74e6d534c0e56defbd1392/databaseUsers/admin/foobar: 404 (request "USER_NOT_FOUND") No user with ID foobar exists.
```

I don't think there should be any concerns about the error return.